### PR TITLE
Enhanced : Batch JSON-RPC logic.

### PR DESCRIPTION
**Pull request by kai890707**
**Description**
Changed scope: 
1. Add the batch rpc request method into `Action`.
2. Changed the way `rpcClient` is called because `rpcClient` was handling `ConcurrentAction` with RPC request obfuscation.
3. Changed `ActionException` logic for `RPC` error.
4. Deleted some error handle method in  `ActionException`.
5. Add new error handle method in  `ActionException`.
6. Changed the README.md and README_zh-TW.md

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] Conforms to style guide

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -231,3 +231,218 @@ var_dump($taichungService->getAction1()->do()->getMeaningData());
...
+    "/"
+))
+->setRpcQuery("/myRpcMethod", [1,2], $id)
+->doneHandler(static function(
+    ResponseInterface $response,
+    Action $runtimeAction
+) {
+    $rpcResponse = $action->getRpcResponse();
+    $result = $action->getRpcResult()[0];
+    $id     = $action->getId()[0];
```
Comment (monkenWu) [inline README.md]: I think `$action->getRpcQueryId()` is better than `$action->getId()`.

File: README.md [hunk 2]
```diff
@@ -231,3 +231,218 @@ var_dump($taichungService->getAction1()->do()->getMeaningData());
...
+    $returnData = [];
+    $rpcResponse = $action->getRpcResponse();
+    $resultArray = $action->getRpcResult();
+    $idArray     = $action->getId();
+    for ($i = 0; $i<3; $i++ ){
+        $returnData[] = [
+            "id" => $idArray[$i],
+            "result" => $resultArray[$i],
+        ]
+    }
```
Comment (monkenWu) [inline README.md:323]: I think the return result of `$action->getRpcResponse();` could be a Key/Value array with `QueryID` as the key.

File: README.md [hunk 3]
```diff
@@ -231,3 +231,218 @@ var_dump($taichungService->getAction1()->do()->getMeaningData());
...
+        "id" => $id,
+        "result" => $result
+    ]);
+})->failHandler(function (
+    ActionException $e
+) {
+    if ($e->isRpcError()) {
+        $rpcResponse = $e->getRpcResponse()
+        $errorResArr = $e->getErrorRpc();
+        $successResArr = $e->getSuccessRpc();
```
Comment (monkenWu) [inline README.md]: I think the return result of both`$e->getErrorRpc();` and `$e->getSuccessRpc();` could be a Key/Value array with `QueryID` as the key.

**Review by monkenWu — COMMENTED** [review]
Viewed

File: README.md (hunk 1 above)
Comment (monkenWu) [reply README.md]: If we use the `batchQuery` pattern, then the return value of `$action->getId()` will become uncertain. I don't think we should provide this method, you should integrate the Query ID with the response structure.

Comment (kai890707) [reply README.md]: Oh~ sorry.
About this, I use getRpcId as the name of the method, the description on README.md is wrong, I will change it in the next commit.

Comment (kai890707) [reply README.md]: So we don't need the `getRpcId()` method, right?
We just need to bind the `rpcid` and `rpcResponse` to the data structure as key-value.

**Comment by kai890707** [comment]
About new commit.
1. In `Action.php`, modify `getRpcResponse()` and `getRpcResult()` to support searching by `RPC ID` and remove the `getRpcId()` method. 
2. In `Action.php`, modify `verifyResponse()` to changing RPC response data structure.
3. In `ActionException.php`, update `getSuccessRpc()` and `getErrorRpc() `to support searching by RPC ID.

File: README.md (hunk 3 above)
Comment (kai890707) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (kai890707) [reply README.md:323]: Done.

File: README.md (hunk 1 above)
Comment (kai890707) [reply README.md]: Done.

**Pull request merged**